rclcpp_action/test/test_actions.cpp

```diff
@@ -0,0 +1,286 @@
+// Copyright 2024 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.#include <gtest/gtest.h>
+
+#include <gtest/gtest.h>
+#include "test_actions.hpp"
+
+struct actions_test_data_t
+{
+    bool use_events_executor;
+    bool use_server_ipc;
+    bool use_client_ipc;
+};
+
+class ActionsTest
+: public testing::Test, public testing::WithParamInterface<actions_test_data_t>
+{
+public:
+    void SetUp() override
+    {
+        test_info = std::make_shared<TestInfo>();
+        rclcpp::init(0, nullptr);
+        auto p = GetParam();
+        std::cout << "Test permutation: "
+                  << (p.use_events_executor ? "{ EventsExecutor, " : "{ SingleThreadedExecutor, ")
+                  << (p.use_server_ipc ? "IPC Server, " : "Non-IPC Server, ")
+                  << (p.use_client_ipc ? "IPC Client }" : "Non-IPC Client }") << std::endl;
+
+        executor = test_info->create_executor(p.use_events_executor);
+        executor_thread = std::thread([&]() {
```

Comment:
include `<thread>`

Reply:
done in 13a27f6f4215f2a539b7e98be6111b0a0b7a0d85